src/backend/gencomm.py

```diff
@@ -45,13 +45,15 @@ def switcher_builtin_element(backend, OUTFILE, blklens, typelist, pupstr, key, v
             yutils.display(OUTFILE, "if (max_nesting_level >= %d) {\n" % nesting_level)
             yutils.display(OUTFILE, "%s->pack = yaksuri_%si_%s_blklen_%s_%s;\n" % (backend, backend, pupstr, blklen, val))
             yutils.display(OUTFILE, "%s->unpack = yaksuri_%si_un%s_blklen_%s_%s;\n" % (backend, backend, pupstr, blklen, val))
+            yutils.display(OUTFILE, "%s->name = \"yaksuri_%si_%s_blklen_%s_%s\";\n" % (backend, backend, pupstr, blklen, val))
```

Comment:
It may be a little confusing here because only the "pack" kernel name is stored. I could store a generic name using "op" instead of "pack".

Reply:
Yes, good idea.